air/src/execution_step/air/canon.rs

```diff
@@ -169,3 +155,17 @@ fn create_canon_stream_from_name(
 
     Ok(result)
 }
+
+/// This function gets a stream from context or return a default empty stream,
+/// it's crucial for deterministic behaviour, for more info see
+/// github.com/fluencelabs/aquavm/issues/346
+fn get_stream_or_default<'ctx, 'value>(
+    ast_canon: &ast::Canon<'_>,
+    exec_ctx: &'ctx ExecutionCtx<'value>,
+) -> Cow<'ctx, Stream> {
+    let maybe_stream = exec_ctx.streams.get(ast_canon.stream.name, ast_canon.stream.position);
```

Comment:
Do you need to pas whole `exec_ctx`, or `streams` is enough?

Reply:
`Streams` is enough, yeah, but it would require importing it into this file